src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -1891,6 +1893,16 @@ internal bool TryGetBytesInternalSequential(int i, byte[] buffer, int index, int
 
             bytesRead = 0;
 
+            // GitHub Issue# 2087 fix, throw error if not accessing fields sequentially
+            bool isSequentialAccess = IsCommandBehavior(CommandBehavior.SequentialAccess);
+            if (isSequentialAccess)
+            {
+                // Track the index of the last column read used to enforce sequential access.
```

Comment:
Why do we need a new variable to track this? don't we already throw in cases where access is not sequential? and if we do how to we detect it?

Reply:
@Wraith2 A new variable was added to track a partially read column.  There is an existing variable to track fully read column value but none for partially read column.